### ISSUE TITLE
[REGRESSION]: PackerBuild task fails with: unknown configuration key: "client_jwt"

### ISSUE THREAD

**Issue by tomasr** [microsoft/azure-pipelines-tasks#19904]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

PackerBuild@1

### Breaking task version

1.239.1

### Last working task version

1.238.1

### Regression Description

We have existing pipelines that used to work and now fail with `unknown configuration key: "client_jwt"`.

I am almost sure this broke due to this change to add support for Workload Identity Federation: https://github.com/microsoft/azure-pipelines-tasks/commit/8947d18848e520e7c706f965418c1fa439932c45

I am not sure if the task version is completely broken, or it's just that it now only works if the azure credentials use WIF.

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows Server 2022

### Relevant log output

```shell
Resolving deploy package path.
Resolved deploy package path: C:\dva\_work\21\b\deploy.
Resolving deploy script path.
Resolved deploy script path: Setup-DriverVM.ps1.
Parsing additional builder parameters json.
Current installed packer version is null.
Either packer is not installed or its version is less than 1.2.4. Version 1.2.4 will be downloaded and installed for running task.
Created staging directory for keeping packer binary and templates: C:\dva\_work\_temp\1716481238825.
Packer installer downloaded successfully at path: C:\dva\_work\_temp\1716481238825\packer.zip.
Packer extracted successfully at path: C:\dva\_work\_temp\1716481238825\packer\packer.exe.
C:\dva\_work\_temp\1716481238825\packer\packer.exe --version
1.2.4

...

Running packer validate command to ensure template is valid
C:\dva\_work\_temp\1716481238825\packer\packer.exe validate -var-file=C:\dva\_work\_temp\1slnxdvoieo.json -var-file=C:\dva\_work\_temp\gixwumfjwme.json C:\dva\_work\_temp\1716481238825\default.managed.windows.template-builderUpdated-fixed.json
Template validation failed. Errors are shown below.

Errors validating build 'azure-arm'. 1 error(s) occurred:

* unknown configuration key: "client_jwt"
##[error]Packer validate command failed. This could happen if task does not support packer version.
```

### Full task logs with system.debug enabled

<details>
  <summary>UNSUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

<details>
  <summary>SUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by KenAdam2024** [comment]
Facing the same issue which has broken down our multiple pipelines, which were running fine before. This is having a major impact on our upcoming releases.

**Comment by tomasr** [comment]
@KenAdam2024 For now I have just revered to running the previous release on my pipelines, like this:

```
  - task: PackerBuild@1.238.1
```

Maybe useful.

**Comment by v-schhabra** [comment]
Hi @KenAdam2024 @tomasr 
Thanks for reporting the issue. As a temporary fix we reverted the task version from 1.239.1 to 1.238.1 
We are just working on the permanent fix. Could you pls check if you can see task version 1.238.1 in your pipelines.
Thank you.

**Comment by v-schhabra** [comment]
@tomasr Could you please share the failed detail logs of the pipeline by adding variable system.debug to "true"
